### PR TITLE
[AAE-4452] FE - [ADF] Fetch destination folder path from mapped variables

### PR DESCRIPTION
**Pull request by sivakumar414ram**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://alfresco.atlassian.net/browse/AAE-4452

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by mauriziovitale — APPROVED** [review]

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts [hunk 1]
```diff
@@ -398,106 +237,126 @@ describe('AttachFileCloudWidgetComponent', () => {
...
+            clickOnAttachButton('attach-file-alfresco');
             fixture.detectChanges();
             await fixture.whenStable();
-            const attachButton: HTMLButtonElement = element.querySelector('#attach-file-alfresco');
+            fixture.detectChanges();
 
-            expect(attachButton).not.toBeNull();
+            const alias = '-root-';
+            const opt = { relativePath: '/pathBasedOnStringvariablevalue' };
+            expect(fetchNodeIdFromRelativePathSpy).toHaveBeenCalledWith(alias, opt);
```
Comment (arditdomi) [inline lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts:254]: let's leave always a space between the test's operations and the expectations. Please fix all the occurrences

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts [hunk 2]
```diff
@@ -57,164 +82,22 @@ describe('AttachFileCloudWidgetComponent', () => {
...
-        },
-        relatedContent: false,
-        contentAvailable: true,
-        link: false,
-        mimeType: 'image/png',
-        simpleType: 'image',
-        previewStatus: 'queued',
-        thumbnailStatus: 'queued'
-    };
+    function clickOnAttachButton(id: string) {
```
Comment (arditdomi) [inline lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts:97]: is this clicking the attach button or the attach widget?

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts [hunk 3]
```diff
@@ -398,106 +237,126 @@ describe('AttachFileCloudWidgetComponent', () => {
...
+            const fetchNodeIdFromRelativePathSpy = spyOn(contentCloudNodeSelectorService, 'fetchNodeIdFromRelativePath');
+            const form = new FormModel({ formVariables, processVariables});
+            createUploadWidgetField(form, 'attach-file-alfresco', [], mockAllFileSourceWithFolderVariablePathType);
             fixture.detectChanges();
             await fixture.whenStable();
-            const attachButton: HTMLButtonElement = element.querySelector('#attach-file-alfresco');
+            clickOnAttachButton('attach-file-alfresco');
+            fixture.detectChanges();
+            await fixture.whenStable();
+            fixture.detectChanges();
```
Comment (arditdomi) [inline lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts:283]: why do you need to call again fixture.detectChanges twice?

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts [hunk 4]
```diff
@@ -398,106 +237,126 @@ describe('AttachFileCloudWidgetComponent', () => {
...
-        it('Should set default user alias (-my-) as rootNodeId if destinationFolderPath does not have alias for Alfresco Content + Locale', async () => {
-            widget.field = new FormFieldModel(new FormModel(), {
-                type: FormFieldTypes.UPLOAD,
-                value: []
-            });
-            widget.field.id = 'attach-file-alfresco';
-            widget.field.params = <FormFieldMetadata> allSourceWithNoAliasParams;
-            widget.field.params.multiple = true;
+        it('Should set default user alias (-my-) as rootNodeId if destinationFolderPath contains wrong alias and multiple upload for Alfresco Content + Locale', async () => {
+
```
Comment (arditdomi) [inline lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts:330]: this space is not needed

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.ts [hunk 5]
```diff
@@ -146,6 +136,28 @@ export class AttachFileCloudWidgetComponent extends UploadCloudWidgetComponent i
             });
     }
 
+    private async getDestinationFolderId() {
```
Comment (arditdomi) [inline lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.ts:139]: can we name it getDestinationFolderNodeId?

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/upload-cloud.widget.ts [hunk 6]
```diff
@@ -160,4 +162,40 @@ export class UploadCloudWidgetComponent extends WidgetComponent implements OnIni
...
+
+    isAlfrescoAndLocal(): boolean {
+        return this.field?.params?.fileSource?.serviceId === FileSourceTypes.ALL_FILE_SOURCES_SERVICE_ID;
+    }
+
+    isPathVariableType(type: string): boolean {
+        return this.field?.params?.fileSource?.destinationFolderPath?.type === type;
+    }
+
+    getDestinationFolderPathFromMappingVariable() {
```
Comment (arditdomi) [inline lib/process-services-cloud/src/lib/form/components/widgets/attach-file/upload-cloud.widget.ts:174]: get is a word that you expect it to return something, can we use a different name other than get?

File: lib/process-services-cloud/src/lib/form/mocks/attach-file-cloud-widget.mock.ts [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+
+export const mockNodeIdBasedOnStringVariableValue = new Promise(function(resolve) {
+    resolve('mock-string-value-node-id');
+});
+
+export const mockNodeIdBasedOnFolderVariableValue = new Promise(function(resolve) {
+    resolve('mock-folder-value-node-id');
+});
+
+export const fakeLocalPngAnswer = {
```
Comment (arditdomi) [inline lib/process-services-cloud/src/lib/form/mocks/attach-file-cloud-widget.mock.ts:193]: can we name it fakeLocalPngResponse instead of Answer?

**Review by arditdomi — CHANGES_REQUESTED** [review]

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.ts (hunk 5 above)
Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.ts]: Updated, Thanks

File: lib/process-services-cloud/src/lib/form/mocks/attach-file-cloud-widget.mock.ts (hunk 7 above)
Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/form/mocks/attach-file-cloud-widget.mock.ts:193]: Updated. Thanks

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts (hunk 4 above)
Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts:318]: Updated, Thanks

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts (hunk 1 above)
Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts:241]: Updated, Thanks

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts (hunk 3 above)
Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts]: Updated, Thanks

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts (hunk 2 above)
Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts]: Updated, Thanks

**Review by mauriziovitale — APPROVED** [review]

**Pull request merged**